applications/StructuralMechanicsApplication/python_scripts/structural_response.py

```diff
@@ -363,4 +374,69 @@ def _SynchronizeAdjointFromPrimal(self):
             adjoint_node.Y = primal_node.Y
             adjoint_node.Z = primal_node.Z
 
+    def _GetAdjointParameters(self):
+
+        adjoint_settings = self.response_settings["adjoint_settings"].GetString()
+
+        if adjoint_settings == "auto":
+            Logger.PrintInfo("\n> Automatic set up adjoint parameters for response:", self.identifier)
+
+            with open(self.response_settings["primal_settings"].GetString(),'r') as parameter_file:
+                primal_parameters = Parameters( parameter_file.read() )
+
+            # check that HDF5 process is not there
+            if primal_parameters["processes"].Has("list_other_processes"):
+                for i in range(0,primal_parameters["processes"]["list_other_processes"].size()):
+                    process = primal_parameters["processes"]["list_other_processes"][i]
+                    raise Exception("Auto setup of adjoint parameters does not support {} in list_other_processes".format(process["python_module"].GetString()))
+
+            # clone primal settings as base for adjoint
+            adjoint_parameters = primal_parameters.Clone()
+
+            # analysis settings
+            solver_settings = adjoint_parameters["solver_settings"]
+            primal_solver_type = solver_settings["solver_type"].GetString()
+            if primal_solver_type != "static":
+                raise Exception("Auto setup of adjoint parameters does not support {} solver_type. Only available for 'static'".format(primal_solver_type))
+            solver_settings["solver_type"].SetString("adjoint_"+primal_solver_type)
+
+            if not solver_settings.Has("compute_reactions"):
+                solver_settings.AddEmptyValue("compute_reactions")
+            solver_settings["compute_reactions"].SetBool(False)
+
+            if not solver_settings.Has("move_mesh_flag"):
+                solver_settings.AddEmptyValue("move_mesh_flag")
+            solver_settings["move_mesh_flag"].SetBool(False)
+
+            if not solver_settings.Has("scheme_settings"):
+                tmp = solver_settings.AddEmptyValue("scheme_settings")
+                if not tmp.Has("scheme_type"):
+                    tmp.AddEmptyValue("scheme_type")
+            solver_settings["scheme_settings"]["scheme_type"].SetString("adjoint_structural")
```

Comment:
those things are quite manual

I think here you could use the assigndefaults-function once that is merged

Reply:
No, here i really have to overwrite the setting in case it already exists.